scripts/scripts.js

```diff
@@ -27,13 +27,28 @@ function buildHeroBlock(main) {
   }
 }
 
+function buildCtaBlock(main) {
+  main.querySelectorAll(':scope > div').forEach((div) => {
+    const h2 = div.querySelector(':scope > h2');
+    const p = div.querySelector(':scope > p');
+    const a = div.querySelector(':scope p > a');
+    // eslint-disable-next-line no-bitwise
+    if (h2 && p && a && (h2.compareDocumentPosition(p) & Node.DOCUMENT_POSITION_FOLLOWING)
+          // eslint-disable-next-line no-bitwise
+          && (p.compareDocumentPosition(a) & Node.DOCUMENT_POSITION_CONTAINED_BY)) {
```

Comment:
I think this check can be avoided by slightly changing your selectors. to `a = p.querySelector('a')` thus ensuring the link is inside the p tag

Reply:
Yes, the decorateButtons function basically adds a class to the p tag parent of the a link. Doesn't do anything outside of that. I am using that for my css though and my CSS is targeting that button only if it is under the CTA section as well:
`main .section.cta`

I did update the check to remove the 5th check and instead go with p.querySelector('a') and check for the existence of p before moving to the larger check. code updated and tested.